src/logger/logreader.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOGREADER_HPP_
+#define LOGREADER_HPP_
+
+#include <aos/common/tools/noncopyable.hpp>
+#include <aos/common/tools/optional.hpp>
+
+#include "logger/types.hpp"
+
+namespace aos::zephyr::logger {
+
+/**
+ * Log entry structure.
+ */
+struct LogEntry {
+    StaticString<cLogEntryLen> mContent;
+    Optional<Time>             mTime = {};
+
+    /**
+     * Resets underlying data.
+     */
+    void Reset()
+    {
+        mContent.Clear();
+        mTime.Reset();
+    }
+};
+
+/**
+ * Log reader interface.
+ */
+class LogReaderItf {
```

Comment:
Is it requried by `LogProvider` only? Probably should be define in log provider header.